### PR TITLE
notifications: Dismiss to watchface when empty

### PR DESCRIPTION
**Pull request by eliedrian**
I found it slightly annoying that dismissing all notifications leaves me with a "No notification to display" message. Instead of dismissing to a relatively useless message, dismiss to watchface.

### CODE REVIEW

**Review by Riksu9000 — COMMENTED** [review]
This seems like an interesting idea. The animation isn't working as I'd expect though. I expect the sideways dismiss animation to go to a black screen and then slide up to the watch face.

File: src/displayapp/screens/Notifications.cpp [hunk 1]
```diff
@@ -112,12 +111,10 @@ void Notifications::Refresh() {
...
                                                        alertNotificationService,
                                                        motorController);
-    } else {
-      currentItem = std::make_unique<NotificationItem>(alertNotificationService, motorController);
-    }
+    }  
   }
 
-  running = currentItem->IsRunning() && running;
+  running = running && currentItem->IsRunning();
```
[the remark this answers is not in the document]
Comment (eliedrian) [reply src/displayapp/screens/Notifications.cpp:117]: I think this is safer. Setting `running` to false implies we're ending the screen. It'd be a waste to create an item to assign to `currentItem` when we're ending anyway.

[the remark this answers is not in the document]
Comment (Riksu9000) [reply src/displayapp/screens/Notifications.cpp:118]: Oh I see. I didn't realize that currentItem is left a nullptr in another function, and removing line 116 means it stays nullptr. Looks like this is more of an issue with code clarity. You should squash the fixup commits and explain these things in the commit message.

File: src/displayapp/screens/Notifications.cpp [hunk 2]
```diff
@@ -82,7 +82,6 @@ void Notifications::Refresh() {
 
   } else if (mode == Modes::Preview && dismissingNotification) {
     running = false;
-    currentItem = std::make_unique<NotificationItem>(alertNotificationService, motorController);
```
[the remark this answers is not in the document]
Comment (Riksu9000) [reply src/displayapp/screens/Notifications.cpp:85]: Well this is for the preview mode, so I believe it's not exactly necessary for this change. It should be explained in the commit description.

**Comment by eliedrian** [comment]
Squashed.

Also updated the comment instead of fully removing it.

**Comment by RageGamerBoi** [comment]
Oh, thank you so much, I've been wanting a feature like this for so long!

**Comment by mark9064** [comment]
Have been daily driving this for months, zero issues

**Comment by mark9064** [comment]
Are there any outstanding issues preventing this from being merged? To me this is an easy usability improvement and as far as I can see all review comments have been addressed

**Comment by eliedrian** [comment]
Yes, anything lacking from this?

**Comment by marigoldfish** [comment]
One more voice in support of this PR being merged :)

**Review by JF002 — APPROVED** [review]

**Review by NeroBurner — APPROVED** [review]
looking good. I like the feature

**Pull request merged**